lib/plugins/aws/package/compile/events/schedule/index.js

```diff
@@ -14,22 +15,61 @@ class AwsCompileScheduledEvents {
       'package:compileEvents': this.compileScheduledEvents.bind(this),
     };
 
-    // TODO: Complete schema, see https://github.com/serverless/serverless/issues/8032
     this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'schedule', {
-      anyOf: [{ type: 'string' }, { type: 'object' }],
+      oneOf: [
+        { type: 'string', pattern: scheduleSyntax },
+        {
+          type: 'object',
+          properties: {
+            rate: { type: 'string', pattern: scheduleSyntax },
+            enabled: { type: 'boolean' },
+            name: { type: 'string', minLength: 1, maxLength: 64, pattern: '[\\.\\-_A-Za-z0-9]+' },
+            description: { type: 'string', maxLength: 512 },
+            input: {
+              oneOf: [
+                { type: 'string', maxLength: 8192 },
+                {
+                  type: 'object',
+                  oneOf: [
+                    {
+                      properties: {
+                        body: { type: 'string', maxLength: 8192 },
+                      },
+                      required: ['body'],
+                      additionalProperties: false,
+                    },
+                    {
+                      not: {
+                        required: ['body'],
+                      },
+                      additionalProperties: true,
```

Comment:
This line is superfluous (by default `additionalProperties` are _true_)

Reply:
Sorry, I do most of my pull requests through Slack :)  Yes, the changes are made.  I put the `additionalProperties: true` in there to make it explicit what the difference between the rules is.

Reply:
Ok, I think same message is send when not defining this property.

Or do you feel we should add `additionalProperties: true` everywhere we allow extra properties, to signal the intent?

Reply:
I think it makes sense where the schema is being relaxed specifically to allow arbitrary objects.  As you say, it signals the intent of the validation.

Reply:
Ok, but in other places we signal that intent by not providing `additionalProperties` at all. Treating this case differently makes it inconsistent, that doesn't seem clean.

Reply:
Sure, there should be some consistency or it's not much use as a signal.  If the convention is to explicitly prevent additional properties unless specifically permitted, I'm perfectly happy to remove it here.

Reply:
It'll be probably cleaner to remove it (I would prefer to keep it consistent and noiseless). If you feel this case deserves such signal, we may express it with comment as its done here: https://github.com/serverless/serverless/blob/51a10aa262af51c5df76e43a804b74c65bba010c/lib/configSchema.js#L33